enhancements/eus-to-eus-upgrade-impediment-reporting-capability.md

```diff
@@ -0,0 +1,209 @@
+---
+title: eus-to-eus-upgrade-impediment-reporting-capability
+
+authors:
+  - "@bentito"
+reviewers:
+  - TBD
+  - "@camilamacedo86"
+  - "@gallettilance"
+approvers:
+  - TBD
+creation-date: 2022-05-17
+last-updated: 2022-05-17
+status: provisional
+see-also:
+  - "/enhancements/none.md"
+replaces:
+  - "/enhancements/none.md"
+superseded-by:
+  - "/enhancements/none.md"
+---
+
+# eus-to-eus-upgrade-impediment-reporting-capability
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+A cluster upgrade between two versions (for instance from one EUS version to
+a later EUS version) may be hampered by critical, operator-delivered, services
+not being able to be upgraded.
+
+## Motivation
+
+Customers using X and Y and Z Red Hat Operators need to know that if they are using
+specific versions of each operator starting at certain OCP version, can they do a 
+successful EUS to EUS upgrade with no manual steps.
+
+
+### Goals
+
+A report showing impediments we want to determine for each operator and display:
+
+1. [Green] Can the operator remain the same version while OCP upgrades around it
+min, max OCP version is encompassed by the proposed OCP version upgrades.
+maxOCPVersion isn’t violated by the proposed upgrades.
+2. [Green] Can the operator be updated (auto-update by OLM) along the same channel as OCP progress through each version.
```

Comment:
A couple of variants:
1.  OpenShift's ability to change channels (e.g. a channel per OCP version)
2. OpenShift's ability to change catalog sources (catalog per OCP version)